### PR TITLE
refactor(lyrics-plus/translator): wait-and-retry logic

### PR DESCRIPTION
**Pull request by pan93412**

### CODE REVIEW

**Review by kyrie25 — APPROVED** [review]
This doesn't solve an issue, merely a rewrite of the same functionality so I'm marking this as a refactor.

File: CustomApps/lyrics-plus/Translator.js [hunk 1]
```diff
@@ -98,10 +98,20 @@ class Translator {
 		}
 	}
 
+	/**
+	 * Async wrapper of `setTimeout`.
+	 *
+	 * @param {number} ms
+	 * @returns {Promise<void>}
+	 */
+	static async #wait(ms) {
```
Comment (rxri) [inline CustomApps/lyrics-plus/Translator.js:107]:
```suggestion
	static async #sleep(ms) {
```
I would also move this function to the bottom of the class

File: CustomApps/lyrics-plus/Translator.js [hunk 2]
```diff
@@ -63,8 +63,8 @@ class Translator {
 			case "ja":
 				if (this.kuroshiro) return;
 				if (typeof Kuroshiro === "undefined" || typeof KuromojiAnalyzer === "undefined") {
-					setTimeout(this.createTranslator.bind(this), 50, lang);
-					return;
+					await Translator.#wait(50);
```
Comment (rxri) [inline CustomApps/lyrics-plus/Translator.js:66]:
```suggestion
					await Translator.#sleep(50);
```

File: CustomApps/lyrics-plus/Translator.js [hunk 3]
```diff
@@ -78,8 +78,8 @@ class Translator {
 			case "ko":
 				if (this.Aromanize) return;
 				if (typeof Aromanize === "undefined") {
-					setTimeout(this.createTranslator.bind(this), 50, lang);
-					return;
+					await Translator.#wait(50);
```
Comment (rxri) [inline CustomApps/lyrics-plus/Translator.js:81]:
```suggestion
					await Translator.#sleep(50);
```

File: CustomApps/lyrics-plus/Translator.js [hunk 4]
```diff
@@ -88,8 +88,8 @@ class Translator {
 			case "zh":
 				if (this.OpenCC) return;
 				if (typeof OpenCC === "undefined") {
-					setTimeout(this.createTranslator.bind(this), 50, lang);
-					return;
+					await Translator.#wait(50);
```
Comment (rxri) [inline CustomApps/lyrics-plus/Translator.js:91]:
```suggestion
					await Translator.#sleep(50);
```

File: CustomApps/lyrics-plus/Translator.js [hunk 5]
```diff
@@ -98,10 +98,20 @@ class Translator {
...
+	 */
+	static async #wait(ms) {
+		return new Promise(resolve => setTimeout(resolve, ms));
+	}
+
 	async romajifyText(text, target = "romaji", mode = "spaced") {
 		if (!this.finished.ja) {
-			setTimeout(this.romajifyText.bind(this), 100, text, target, mode);
-			return;
+			await Translator.#wait(100);
```
Comment (rxri) [inline CustomApps/lyrics-plus/Translator.js:113]:
```suggestion
			await Translator.#sleep(100);
```

File: CustomApps/lyrics-plus/Translator.js [hunk 6]
```diff
@@ -112,8 +122,8 @@ class Translator {
 
 	async convertToRomaja(text, target) {
 		if (!this.finished.ko) {
-			setTimeout(this.convertToRomaja.bind(this), 100, text, target);
-			return;
+			await Translator.#wait(100);
```
Comment (rxri) [inline CustomApps/lyrics-plus/Translator.js:125]:
```suggestion
			await Translator.#sleep(100);
```

File: CustomApps/lyrics-plus/Translator.js [hunk 7]
```diff
@@ -122,8 +132,8 @@ class Translator {
 
 	async convertChinese(text, from, target) {
 		if (!this.finished.zh) {
-			setTimeout(this.convertChinese.bind(this), 100, text, target);
-			return;
+			await Translator.#wait(100);
```
Comment (rxri) [inline CustomApps/lyrics-plus/Translator.js:135]:
```suggestion
			await Translator.#sleep(100);
```

**Review by rxri — CHANGES_REQUESTED** [review]

**Review by rxri — APPROVED** [review]

**Pull request merged**